src/CoreText/CTFontManager.cs

```diff
@@ -267,5 +349,168 @@ public static NSObject ObserveRegisteredFontsChanged (EventHandler<NSNotificatio
+				return null;
+			// Copy/Create rule - dont't retain it inside the .ctor
+			return new CTFontDescriptor (p, owns: false);
+		}
+
+		[Watch (6,0), TV (13,0), Mac (10,15, onlyOn64: true), iOS (13,0)]
+		[DllImport (Constants.CoreTextLibrary)]
+		static extern unsafe /* CFArrayRef */ IntPtr CTFontManagerCreateFontDescriptorsFromData (/* CFDataRef */ IntPtr data);
+
+		[Watch (6,0), TV (13,0), Mac (10,15, onlyOn64: true), iOS (13,0)]
+		public static CTFontDescriptor[] CreateFontDescriptors (NSData data)
+		{
+			if (data == null)
+				throw new ArgumentNullException (nameof (data));
+
+			var p = CTFontManagerCreateFontDescriptorsFromData (data.Handle);
+			try {
+				return NSArray.ArrayFromHandle<CTFontDescriptor> (p);
+			}
+			finally {
+				// Copy/Create rule - we must release the CFArrayRef
+				CFObject.CFRelease (p);
+			}
+		}
+
+#if __IOS__
+		[NoWatch, NoTV, NoMac, iOS (13,0)]
+		[DllImport (Constants.CoreTextLibrary)]
+		static extern unsafe void CTFontManagerRegisterFontsWithAssetNames (/* CFArrayRef */ IntPtr fontAssetNames, /* CFBundleRef _Nullable */ IntPtr bundle, CTFontManagerScope scope, bool enabled, void * registrationHandler);
+
+		// reminder that NSBundle and CFBundle are NOT toll-free bridged :(
+		[NoWatch, NoTV, NoMac, iOS (13,0)]
+		public static void RegisterFonts (string[] assetNames, CFBundle bundle, CTFontManagerScope scope, bool enabled, CTFontRegistrationHandler registrationHandler)
+		{
+			using (var arr = EnsureNonNullArray (assetNames, nameof (assetNames))) {
+				unsafe {
+					if (registrationHandler == null) {
+						CTFontManagerRegisterFontsWithAssetNames (arr.Handle, bundle.GetHandle (), scope, enabled, null);
+						return;
+					}
+					BlockLiteral *block_ptr_handler;
+					BlockLiteral block_handler = new BlockLiteral ();
+					block_ptr_handler = &block_handler;
+					block_handler.SetupBlock (callback, registrationHandler);
+					CTFontManagerRegisterFontsWithAssetNames (arr.Handle, bundle.GetHandle (), scope, enabled, (void*) block_ptr_handler);
+					block_ptr_handler->CleanupBlock ();
+				}
+			}
+		}
+
+		[NoWatch, NoTV, NoMac, iOS (13,0)]
+		public delegate void CTFontManagerRequestFontsHandler (CTFontDescriptor[] unresolvedFontDescriptors);
+
+		[NoWatch, NoTV, NoMac, iOS (13,0)]
+		[DllImport (Constants.CoreTextLibrary)]
+		static extern unsafe void CTFontManagerRequestFonts (/* CFArrayRef */ IntPtr fontDescriptors, void * completionHandler);
+
+		internal delegate void InnerRequestFontsHandler (IntPtr block, IntPtr fontDescriptors);
+		static readonly InnerRequestFontsHandler requestCallback = TrampolineRequestFonts;
+
+		[MonoPInvokeCallback (typeof (InnerRequestFontsHandler))]
+		static unsafe void TrampolineRequestFonts (IntPtr block, /* CFArray */ IntPtr fontDescriptors)
+		{
+			var descriptor = (BlockLiteral *) block;
+			var del = (CTFontManagerRequestFontsHandler) (descriptor->Target);
+			if (del != null)
+				del (NSArray.ArrayFromHandle<CTFontDescriptor> (fontDescriptors));
```

Comment:
I wonder if we would like to clean (release) the `CTFontDescriptor`s individually after calling `del` with them, at least it is not a `CMSampleBuffer []` :P

Reply:
code can keep reference to them, e.g in fields
`CMSampleBuffer` is not a good example and requires extra care from callers (no nice way to handle it)